File: frontend/src/types/project.ts
export interface Project {
  id: number
  name: string
  description: string
  package_path: string
  created_at: string
  owner_id: number
  status: 'DEVELOPING' | 'ONLINE' | 'OFFLINE'
  version: string
  entrypoint: string
  has_requirements: boolean
  env_template: Record<string, string> | null
}

export interface ProjectCreate {
  name: string
  description: string
  file?: File
  files?: File[]
  git_repo_url?: string
  git_branch?: string
  ssh_key?: string
  ssh_key_fingerprint?: string
}

export interface ProjectUpdate {
  name?: string
  description?: string
  status?: 'DEVELOPING' | 'ONLINE' | 'OFFLINE'
  version?: string
  entrypoint?: string
  env_template?: Record<string, string>
}

// Git 凭证相关类型
export interface GitCredential {
  id: number
  provider: string
  username: string
  created_at: string
  // 新增SSH密钥相关字段
  has_ssh_key: boolean
  ssh_key_fingerprint?: string
}

export interface GitCredentialCreate {
  provider: string
  username: string
  token: string
  // 新增SSH密钥相关字段
  ssh_private_key?: string
  ssh_key_fingerprint?: string
}

export interface GitCredentialUpdate {
  provider?: string
  username?: string
  token?: string
  // 新增SSH密钥相关字段
  ssh_private_key?: string
  ssh_key_fingerprint?: string
}

File: backend/app/schemas/project.py
# schemas/project.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime


class ProjectBase(BaseModel):
    name: str
    description: Optional[str] = None


class ProjectCreate(ProjectBase):
    # 前端不需要传 owner_id，由后端自动注入
    owner_id: Optional[int] = None
    package_path: Optional[str] = None


class ProjectUpdate(ProjectBase):
    pass


class Project(ProjectBase):
    id: int
    package_path: str
    created_at: datetime
    owner_id: int  # 返回时可以包含 owner_id
    status: str
    version: Optional[str] = None
    entrypoint: str
    has_requirements: bool
    env_template: Optional[dict] = None

    class Config:
        from_attributes = True


class ProjectOut(ProjectBase):
    id: int
    package_path: Optional[str] = None
    created_at: datetime
    owner_id: int

    class Config:
        from_attributes = True
